packages/ii-login-button/src/ii-login-button.ts

```diff
@@ -0,0 +1,277 @@
+    const login = this._login.bind(this);
+    newButton.addEventListener("click", login);
+  }
+
+  attributeChangedCallback(attrName: string, oldVal: unknown, newVal: unknown) {
+    const button = this.shadowRoot?.getElementById(
+      "ii-login-button"
+    ) as HTMLButtonElement;
+    switch (attrName) {
+      case "disabled": {
+        button.disabled = newVal === "true";
+        break;
+      }
+      case "label": {
+        (
+          button.querySelector("#ii-login-button-label") as HTMLSpanElement
+        ).innerText = newVal as string;
+        break;
+      }
+      case "innerstyle": {
+        button.setAttribute("style", newVal as string);
+        break;
+      }
+      default: {
+        console.log("unhandled attribute change", attrName, oldVal, newVal);
+        break;
+      }
+    }
+  }
+
+  static get observedAttributes() {
+    return ["onSuccess", "label", "disabled", "innerstyle"];
+  }
+
+  /**
+   * @returns the internal {@link AuthClient} used in the component
+   */
+  get authClient(): AuthClient {
+    if (!this._state.authClient) {
+      throw new Error("authClient has failed to initialize");
+    }
+    return this._state.authClient;
+  }
+
+  /**
+   * @returns whether the current identity is authenticated or is anonymous
+   */
+  get isAuthenticated(): boolean {
+    return this._state.isAuthenticated;
+  }
+
+  /**
+   * @returns the current {@link Identity}
+   */
+  get identity(): Identity | undefined {
+    return this._state.identity;
+  }
+
+  /**
+   * @returns the {@link Principal} of the current {@link Identity}
+   */
+  get principal(): Principal | undefined {
+    return this._state.identity?.getPrincipal();
+  }
+
+  /**
+   * @returns string of the {@link Principal}
+   */
+  get principalString(): string | undefined {
+    return this.principal?.toString();
+  }
+
+  /**
+   * @returns the default {@link AccountIdentifier} for the current principal as a string
+   */
+  get accountId(): string | undefined {
+    if (this.principal) {
+      const id = AccountIdentifier.fromPrincipal({ principal: this.principal });
+      return id.toHex();
+    }
+    return undefined;
+  }
+
+  /**
+   * internal method - do not call
+   */
+  private _login(): void {
+    const handleSuccess = this._handleSuccess.bind(this);
+    this.authClient.login({
+      ...this._state.loginOptions,
+      onSuccess: handleSuccess,
+    });
+  }
+
+  /**
+   * internal method - do not call
+   */
+  private async _handleSuccess() {
+    this.setState({
+      authClient: this.authClient,
+      isAuthenticated: await this.authClient.isAuthenticated(),
+      identity: await this.authClient.getIdentity(),
+    });
+    const event = new LoginEvent("login", { bubbles: true, composed: true });
+    this._state.loginOptions?.onSuccess?.();
+    this.dispatchEvent(event);
+  }
+
+  /**
+   * Logs out authClient and clears storage
+   */
+  public logout() {
+    this.setState({
+      isAuthenticated: false,
```

Comment:
can `isAuthenticated` and `identity` become desynchronized in the state - would it be worth to keep only the `indentity` in state and interfer the authenticated information?

Reply:
At this point they shouldn't be able to get out of sync, and I don't expect the complexity of this component to grow much more. I figure the convenience is worth it

Reply:
I see. Works me me, thanks for the feedback.